bin/ask_update.py

```diff
@@ -56,6 +56,24 @@ def get_all_known_facts():
             %s
         }
 
+    @cacheit
+    def get_all_known_matrix_facts():
+        """
+        Known facts between unary predicates for matrices as CNF clauses.
+        """
+        return {
+            %s
+        }
+
+    @cacheit
+    def get_all_known_nonmatrix_facts():
+        """
+        Known facts between unary predicates for numbers as CNF clauses.
+        """
+        return {
+            %s
+        }
```

Comment:
I think that distinguishing facts as "matrix facts" plus "non matrix facts" is not very scalable. Probably it would be better here to distinguish "matrix facts" and "number facts". Ideally this would tied to the `.kind` of the expression:
```
In [9]: Matrix([[1, 2], [3, 4]]).kind
Out[9]: MatrixKind(NumberKind)

In [10]: sqrt(2).kind
Out[10]: NumberKind
```